packages/admin/admin/src/clipboard/readClipboardText.ts

```diff
@@ -7,11 +7,32 @@ export async function readClipboardText(): Promise<string | undefined> {
         return undefined;
     }
 
+    // Firefox does not support navigator.clipboard.readText() by default.
+    if (navigator.clipboard.readText === undefined) {
+        return window.localStorage.getItem("comet_clipboard") ?? undefined;
+    }
+
     if (canReadClipboard === true) {
         return navigator.clipboard.readText();
     }
 
-    const permissionStatus = await navigator.permissions.query({ name: "clipboard-read" as PermissionName });
+    let permissionStatus: PermissionStatus;
+
+    try {
+        permissionStatus = await navigator.permissions.query({ name: "clipboard-read" as PermissionName });
+    } catch (error) {
+        // Firefox with "dom.events.asyncClipboard.readText" enabled via about:config. navigator.clipboard.readText() is available, but permission
+        // 'clipboard-read' permission cannot be queried. We can still read the clipboard though.
+        if (
+            error instanceof TypeError &&
+            error.message === "'clipboard-read' (value of 'name' member of PermissionDescriptor) is not a valid value for enumeration PermissionName."
```

Comment:
This is the error message Firefox shows when querying the permission. Should we make this check more resilient to changes? For instance by `error.message.includes("clipboard-read")` or `/clipboard-read.*is.*not.*valid`.test(error.message)`. Is this too generic or too specific?

Reply:
I'm not sure if the error-message could be in a different language, depending on the browser/os language 🤔
If it's certainly always english, using `error.message.includes("clipboard-read")` is probably a good idea. 

Alternatively you could maybe just call `navigator.clipboard.readText()` and then use session-storage, if that throws an error.